### DISCUSSION TITLE
Is it possible to defuse tumblr.com click events?

### DISCUSSION THREAD

**Discussion by wavygecko** [uBlockOrigin/uAssets#14605] [Q&A] [answered]
I'm trying to defuse `click` events on `tumblr.com` but for some reason it doesn't seem to be working.

In devtools I see the `click` event I want to defuse:
<img width="571" alt="image" src="https://user-images.githubusercontent.com/110956608/187603511-46a0c44e-daf3-45ac-9527-32bc40502f5b.png">

But `tumblr.com##+js(aeld, click)` does nothing and `tumblr.com##+js(aell)` shows no console logs.

Any idea what I'm doing wrong?

re: https://github.com/DandelionSprout/adfilt/discussions/642

**Comment by gorhill** [comment]
In Firefox, uBO's injected scriptlets are affected by a page's current CSP, and `tumblr.com` forbids inline scripts, which prevents uBO's injected scriptlet from being executed.

**Reply by wavygecko** [reply]
Does Chrome have less strict CSP enforcement?

**Reply by gorhill** [reply]
It seems Chromium properly exclude extension-injected inline scripts. Extensions extend browser capabilities, so it makes sense that they are not restricted by webpages.

**Reply by MasterKia** [reply]
@gorhill [You said](https://github.com/gorhill/uBlock/issues/2823#issuecomment-317943668): 
> I don't plan to have uBO lower CSP rules set by a site. In my opinion, being able to inject a ~~surrogate~~ scriptlet is less important than not relaxing existing CSP ruleset.

Does this mean sites can trivially [gain _protection_](https://github.com/uBlockOrigin/uAssets/issues/14204) against uBO's `+js` rules in Firefox?
With the rise of MV3 more users would switch from Chromium to Firefox.

**Reply by gorhill** [reply]
I need to investigate what can be done with Firefox, there [might be a way](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Sharing_objects_with_page_scripts). Also I am pretty sure there is an open issue in bugzilla about this, which I do not have the time to try to find now.

**Reply by MasterKia** [reply]
> [I am pretty sure](https://github.com/uBlockOrigin/uBlock-issues/issues/2079#issuecomment-1088692802) there is an open issue in bugzilla about this

A 6 years old one:
https://bugzilla.mozilla.org/show_bug.cgi?id=1267027

**Comment by gorhill** [comment, marked as answer]
Turns out it's possible to bypass a page's CSP, see <https://github.com/uBlockOrigin/uBlock-issues/issues/235#issuecomment-1344313926>. Should be solved in 1.45.3b14+.
